src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspace.cs

```diff
@@ -34,6 +41,22 @@ internal VisualStudioWorkspace(HostServices hostServices, WorkspaceBackgroundWor
             }
         }
 
+        private void OnCacheFlushRequested(object sender, EventArgs e)
+        {
+            if (_backgroundCompiler != null)
+            {
+                _backgroundCompiler.Dispose();
+                _backgroundCompiler = null; // PartialSemanticsEnabled will now return false
```

Comment:
If data tips are the only thing using it, I think we did that because in practice it only requires your current syntax tree + metadata references to mscorlib. In that case, we don't need to turn off partial semantics in this PR, since the features we thought needed the background compiler don't.